lockc/src/utils.rs

```diff
@@ -0,0 +1,35 @@
+use std::io;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum HashError {
+    #[error("could not convert the hash to a byte array")]
+    ByteWriteError(#[from] io::Error),
+}
+
+/// Simple string hash function which allows to use strings as keys for BPF
+/// maps even though they use u32 as a key type.
+pub fn hash(s: &str) -> Result<u32, HashError> {
+    let mut hash: u32 = 0;
+
+    for c in s.chars() {
+        let c_u32 = c as u32;
+        hash += c_u32;
+    }
+
+    Ok(hash)
+}
```

Comment:
Silly questions:

 * how can this return an Error?
 * are we sure there could be no hashing collisions?